### PR TITLE
Some minor HTMLMediaElement improvements

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18504)

### CODE REVIEW

File: components/script/dom/htmlmediaelement.rs [hunk 1]
```diff
@@ -835,22 +856,22 @@ impl FetchResponseListener for HTMLMediaElementContext {
...
             elem.upcast::<EventTarget>().fire_event(atom!("progress"));
 
-            elem.network_state.set(NETWORK_IDLE);
+            elem.network_state.set(NetworkState::Idle);
 
             elem.upcast::<EventTarget>().fire_event(atom!("suspend"));
         }
         // => "If the connection is interrupted after some media data has been received..."
-        else if elem.ready_state.get() != HAVE_NOTHING {
+        else if elem.ready_state.get() != ReadyState::HaveEnoughData {
```
Comment (emilio) [inline components/script/dom/htmlmediaelement.rs]: This should be `HaveNothing`, right? Or is this change intentional?

File: components/script/dom/htmlmediaelement.rs [hunk 2]
```diff
@@ -618,9 +639,9 @@ impl HTMLMediaElementMethods for HTMLMediaElement {
             task_source.queue_simple_event(self.upcast(), atom!("play"), &window);
 
             // 7.4
-            if state == HAVE_NOTHING ||
-               state == HAVE_METADATA ||
-               state == HAVE_CURRENT_DATA {
+            if state == ReadyState::HaveEnoughData ||
+               state == ReadyState::HaveEnoughData ||
```
Comment (emilio) [inline components/script/dom/htmlmediaelement.rs]: This should be HaveMetadata.

File: components/script/dom/htmlmediaelement.rs (hunk 1 above)
Comment (nox) [reply components/script/dom/htmlmediaelement.rs]: REPLACING IS HARD

File: components/script/dom/htmlmediaelement.rs [hunk 3]
```diff
@@ -507,8 +528,8 @@ impl HTMLMediaElement {
             // TODO 4.4 (forget resource tracks)
 
             // 4.5
-            if self.ready_state.get() != HAVE_NOTHING {
-                self.change_ready_state(HAVE_NOTHING);
+            if self.ready_state.get() != ReadyState::HaveEnoughData {
```
Comment (emilio) [inline components/script/dom/htmlmediaelement.rs]: Why HaveEnughData instead of HaveNothing?

**Review by emilio — APPROVED** [review]
r=me with those fixed.
